pkg/api/util/doget.go

```diff
@@ -76,3 +76,34 @@ func DoPost(c *http.Client, url string, contentType string, body io.Reader) (res
 	}
 	return resp, nil
 }
+
+// DoPostChunked is a wrapper around performing HTTP GET requests that stream chunked data
+func DoPostChunked(c *http.Client, url string, contentType string, body io.Reader, onChunk func([]byte)) error {
+	req, e := http.NewRequest("POST", url, body)
+	if e != nil {
+		return e
+	}
+	req.Header.Set("Content-Type", contentType)
+	req.Header.Set("Authorization", "Bearer "+GetAuthToken())
+
+	r, e := c.Do(req)
+	if e != nil {
+		return e
+	}
+	defer r.Body.Close()
+
+	var m int
+	for {
+		buf := make([]byte, 128)
```

Comment:
Can we move this allocation outside the loop?

Reply:
I need a fresh buffer on every iteration. Is there a better way?

Reply:
Fresh in the sense of handling the read of `n` < `len(buf)`? You do get the `n` of bytes read from `Read` to handle that.

Reply:
you are totally right 🤦‍♂️ my bad. 
I updated it to use `buf[:m]` - thanks!